pkg/virt-api/webhooks/defaults.go

```diff
@@ -50,9 +50,20 @@ func SetDefaultVirtualMachineInstance(clusterConfig *virtconfig.ClusterConfig, v
 	setDefaultHypervFeatureDependencies(&vmi.Spec)
 	setDefaultCPUArch(clusterConfig, &vmi.Spec)
 	setGuestMemoryStatus(vmi)
+	setCurrentCPUTopologyStatus(vmi)
 	return nil
 }
 
+func setCurrentCPUTopologyStatus(vmi *v1.VirtualMachineInstance) {
```

Comment:
nit - Have you considered moving this function to the VMI controller?
In my opinion `.status` fields are usually set by the controller that watches the resource.

For example here: https://github.com/kubevirt/kubevirt/blob/e6207f255d85c24567f1d28b5a6f232279ae68ff/pkg/virt-controller/watch/vmi.go#L533

Reply:
Yeah I think moving both this and `setGuestMemoryStatus` fully over to the VMI controller is valid even for this initial state capture the first time the VMI is seen. I'll draft a follow up shortly for this.

Reply:
any reasons not to do this immediately? Moving it to the webhook and the back to the controller seems wrong to me

Reply:
@alicefr apologies missed this yesterday, I don't have the bandwidth to include this in the series before 1.3 feature freeze. Note that this isn't moving the code back to the same VM controller but the VMI controller so I think it's valid to defer.